io/module_write_netcdf.F90

```diff
@@ -172,13 +177,15 @@ subroutine write_netcdf(fieldbundle, wrtfb, filename, mpi_comm, mype, im, jm, rc
         end if
       else if (fldlev(i) > 1) then
          if (typekind == ESMF_TYPEKIND_R4) then
-           ncerr = nf90_def_var(ncid, trim(fldName), NF90_FLOAT, &
-                                (/im_dimid,jm_dimid,pfull_dimid,time_dimid/), varids(i)); NC_ERR_STOP(ncerr)
            if (ideflate > 0) then
-              ! shuffle filter off since lossy compression used
-              ncerr = nf90_def_var_deflate(ncid, varids(i), 0, 1, ideflate)
-              NC_ERR_STOP(ncerr)
-           endif
+             ncerr = nf90_def_var(ncid, trim(fldName), NF90_FLOAT, &
+                    (/im_dimid,jm_dimid,pfull_dimid,time_dimid/), varids(i), &
+                    shuffle=.false.,deflate_level=ideflate,&
+                    chunksizes=(/im,jm,1,1/),cache_size=40*im*jm); NC_ERR_STOP(ncerr)
```

Comment:
Must the chunksizes be /im,jm,1,1/? 

Having chunksizes of 1 performs badly. Having chunksizes that are the entire extent of the dimension can be fast for smaller sizes, but larger sizes should be broken up.

Reply:
No, I just set it to that because it seemed like the most common access pattern (reading a 2-d horizontal grid).